src/classes/RD_RecurringDonations_TEST2.cls

```diff
@@ -531,68 +531,74 @@ private with sharing class RD_RecurringDonations_TEST2 {
     }
 
     //--------------- from RD_RecurringDonations_BATCH -----------------
-    @isTest 
+    @isTest
     public static void testBatch() {
-        RD_RecurringDonations_TEST.createRdSettingsWithYearForecast();           
-                    
+        RD_RecurringDonations_TEST.createRdSettingsWithYearForecast();
+
         Account a = buildAccount();
         insert a;
-        
+
         Contact c = buildContact(a.Id);
-        insert c;       
+        insert c;
 
         npe03__Recurring_Donation__c rd = RD_RecurringDonations_TEST.buildRecurringDonationWeeklyInstallment(null);
         rd.npe03__Organization__c = a.id;
         rd.npe03__Open_Ended_Status__c = System.Label.npe03.RecurringDonationOpenStatus;
         rd.npe03__Next_Payment_Date__c = TODAY;
         insert rd;
-        
+
         List<Opportunity> originalOpps = RD_RecurringDonations_TEST.getOppsOrderByCloseDate(rd.Id);
-        
+
         Opportunity o = originalOpps[0];
         o.StageName = UTIL_UnitTestData_TEST.getClosedWonStage();
         o.CloseDate = TODAY;
-        update o;        
-        
+        update o;
+
         //kick it off from the vf page
         Test.setCurrentPageReference(new PageReference('Page.RecurringDonationsSettings'));
-        
+
         Test.StartTest();
         RD_RecurringDonations_BATCH rdbatch = new RD_RecurringDonations_BATCH();
         ID ApexJobId = Database.executeBatch(rdbatch, 10);
         Test.stopTest();
-    }  
-    
+    }
+
     /***
      * @description Verifies Recurring Donations are returned when enhanced Recurring Donations are not enabled
      */
-    @isTest 
-    private static void shouldReturnRDsWhenEnhancementIsNotEnabled() {  
-        List<npe03__Recurring_Donation__c> records = executeBatchStart();
+    @isTest
+    private static void shouldReturnRDsWhenEnhancementIsNotEnabled() {
+        List<npe03__Recurring_Donation__c> records = new List<npe03__Recurring_Donation__c>();
+        try {
+            records = executeBatchStart();
+        } catch (Exception ex) { }
+
+        System.assertEquals(1, records.size(), 'Recurring Donation record should be returned from the start() method');
+    }
 
-        System.assertEquals(1, records.size(), 'Recurring Donation record should be returned from the start() method');       
-    } 
-    
     /***
      * @description Verifies Recurring Donations are not returned when enhanced Recurring Donations are enabled
      */
-    @isTest 
-    private static void shouldNotReturnRDsWhenEnhancementIsEnabled() {  
+    @isTest
+    private static void shouldNotReturnRDsWhenEnhancementIsEnabled() {
         RD2_EnablementService_TEST.setRecurringDonations2Enabled();
 
-        List<npe03__Recurring_Donation__c> records = executeBatchStart();
+        List<npe03__Recurring_Donation__c> records = new List<npe03__Recurring_Donation__c>();
```

Comment:
The test method name should verify the exception is thrown in this case now. We will not return zero records any more.

Reply:
@force2b ^